firebase-perf/src/main/java/com/google/firebase/perf/application/FrameMetricsRecorder.java

```diff
@@ -107,11 +108,22 @@ public Optional<PerfFrameMetrics> stop() {
     }
     Optional<PerfFrameMetrics> data = this.snapshot();
     try {
+      // No reliable way to check for hardware-acceleration, so we must catch retroactively (#2736).
       frameMetricsAggregator.remove(activity);
-    } catch (IllegalArgumentException err) {
+    } catch (IllegalArgumentException | NullPointerException ex) {
+      // Both of these exceptions result from android.view.View.addFrameMetricsListener silently
+      // failing when the view is not hardware-accelerated. Successful addFrameMetricsListener
+      // stores an observer in a list, and initializes the list if it was uninitialized. Invoking
+      // View.removeFrameMetricsListener(listener) throws IAE if it doesn't exist in the list, or
+      // throws NPE if the list itself was never initialized (#4184).
+      if (ex instanceof NullPointerException && Build.VERSION.SDK_INT > Build.VERSION_CODES.P) {
+        // Re-throw above API 28, since the NPE is fixed in API 29:
+        // https://android.googlesource.com/platform/frameworks/base/+/140ff5ea8e2d99edc3fbe63a43239e459334c76b
+        throw ex;
```

Comment:
Is the intent to notify the developers of the exception during development? Interesting to see a pattern where we throw and exception in a catch module.